tests/integration/steps/integration_steps.go

```diff
@@ -0,0 +1,699 @@
+	"github.com/gruntwork-io/terratest/modules/random"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var namespace = "k8ssandra"
+var repairId string
+
+func RunShellCommand(command *exec.Cmd) error {
+	err := command.Start()
+	if err != nil {
+		log.Fatal(err)
+	}
+	err = command.Wait()
+	return err
+}
+
+func RunShellCommandAndGetOutput(command *exec.Cmd) string {
+	var outb bytes.Buffer
+	command.Stdout = &outb
+	err := command.Run()
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	return string(outb.String())
+}
+
+func GetServicesWithLabel(label string) (*v1.ServiceList, error) {
+	kubectlOptions := k8s.NewKubectlOptions("", "", namespace)
+	clientset, err := k8s.GetKubernetesClientFromOptionsE(GinkgoT(), kubectlOptions)
+	Expect(err).To(BeNil())
+	services, err := clientset.CoreV1().Services(namespace).List(context.Background(), metav1.ListOptions{LabelSelector: label})
+	return services, err
+}
+
+// Find returns true if val exists in the slice array, false otherwise
+func Find(slice []string, val string) bool {
+	for _, item := range slice {
+		if item == val {
+			return true
+		}
+	}
+	return false
+}
+
+func deployCluster(customValues string, helmValues map[string]string) {
+	clusterChartPath, err := filepath.Abs("../../charts/k8ssandra")
+	if err != nil {
+		Fail("Couldn't find the absolute path for K8ssandra charts")
+	}
+
+	customChartPath, err := filepath.Abs("charts/" + customValues)
+	if err != nil {
+		Fail(fmt.Sprintf("Couldn't find the absolute path for custom values: %s", customValues))
+	}
+
+	kubectlOptions := k8s.NewKubectlOptions("", "", namespace)
+	helmOptions := &helm.Options{}
+
+	helmOptions = &helm.Options{
+		// Enable traefik to allow redirections for testing
+		SetValues:      helmValues,
+		KubectlOptions: k8s.NewKubectlOptions("", "", namespace),
+		ValuesFiles:    []string{customChartPath},
+	}
+
+	releaseName := "k8ssandra"
+	defer timeTrack(time.Now(), fmt.Sprintf("Installing and starting k8ssandra"))
+	helm.Install(GinkgoT(), helmOptions, clusterChartPath, releaseName)
+
+	// Wait for cass-operator pod to be ready
+	attempts := 0
+	maxAttempts := 10
+	for {
+		attempts++
+		clientset, _ := k8s.GetKubernetesClientFromOptionsE(GinkgoT(), kubectlOptions)
+		pods, _ := clientset.CoreV1().Pods(namespace).List(context.Background(), metav1.ListOptions{LabelSelector: "app.kubernetes.io/name=cass-operator"})
+		if len(pods.Items) == 1 {
+			k8s.RunKubectl(GinkgoT(), kubectlOptions, "wait", "--for=condition=Ready", "pod", "-l", "app.kubernetes.io/name=cass-operator", "--timeout=1800s")
+			break
+		} else if attempts > maxAttempts {
+			Fail("Couldn't find cass-operator pod")
+		}
+		time.Sleep(20 * time.Second)
+	}
+
+	// Wait for CassandraDatacenter to be ready..
+	k8s.RunKubectl(GinkgoT(), kubectlOptions, "wait", "--for=condition=Ready", "cassandradatacenter/dc1", "--timeout=1800s")
```

Comment:
Your suggested alternative looks more verbose, unless I'm missing something.
Could you elaborate on how using apimachinery is better than relying on `kubectl wait`?
Is it because you'd like to remove Terratest?

Reply:
Using `wait` or `Eventually` is definitely more verbose but it allows us to log updates which I find helpful when polling for a long time. The particular example I referenced unfortunately does not do any logging, but this [one](https://github.com/k8ssandra/reaper-operator/blob/master/test/framework/framework.go#L159) does.